### DISCUSSION TITLE
Improve form width options

### DISCUSSION THREAD

**Discussion by grantfitzsimmons** [specify/specify7#5887] [Feature Requests]
It would be great if we could set the form width (in the definition) to occupy a set percentage of the screen with certain breakpoints, or add some other way to make the forms wider so that they do not collapse and become extremely narrow when subviews are not open.

> custom.views.xml: CollectionObject: Added spacer
> S7 has rendering issues…
> If the form has subviews on it but they are in an empty state [blank form - new CO], then S7 renders the entire form very narrow.
> Adding this spacer to the bottom of the CO form makes it appear wider and more UX-friendly.

<img width="668" alt="image" src="https://github.com/specify/specify7/assets/37256050/8ac43165-3111-4eda-ad1f-d4ad064a2090">

**Requested By:** Matthew Cruz at The University of Michigan

**Comment by maxpatiiuk** [comment]
Ah, the spacer is a neat workaround!
The other one is using very high colspan, and then putting entire form into a panel

note than in sp7 you don't have to define columns to use those columns so this workaround is only practical in sp7, not sp6

**Comment by maxpatiiuk** [comment]
The other solution is to disable "Flexible form width" (or how it's called) in preferences - that will make form column sizing match sp6 - based on pixels not percentages

but yeah, would probably be nice to add a base "form width" attribute, that then is respected by all columns
that would even resolve some of the issues we keep on having with form size changing when going between records in a dialog (at a cost of forcing more work on collection managers as far as maintaining forms, and possibly reducing adaptability of forms)

**Comment by maxpatiiuk** [comment]
Maybe instead of width then it should be min-widh?
